pkg/clusteragent/admission/mutate/autoscaling/autoscaling.go

```diff
@@ -27,23 +29,25 @@ const (
 
 // Webhook implements the MutatingWebhook interface
 type Webhook struct {
-	name       string
-	isEnabled  bool
-	endpoint   string
-	resources  []string
-	operations []admiv1.OperationType
-	patcher    workload.PodPatcher
+	name        string
+	webhookType common.WebhookType
+	isEnabled   bool
+	endpoint    string
+	resources   []string
+	operations  []admissionregistrationv1.OperationType
+	patcher     workload.PodPatcher
 }
 
 // NewWebhook returns a new Webhook
 func NewWebhook(patcher workload.PodPatcher) *Webhook {
 	return &Webhook{
-		name:       webhookName,
-		isEnabled:  config.Datadog().GetBool("autoscaling.workload.enabled"),
-		endpoint:   webhookEndpoint,
-		resources:  []string{"pods"},
-		operations: []admiv1.OperationType{admiv1.Create},
-		patcher:    patcher,
+		name:        webhookName,
+		webhookType: common.MutatingWebhook,
```

Comment:
Since this field seems to always have the same value, it may not need to be a field of this struct.

```suggestion
```